addons/mail_group/controllers/main.py

```diff
@@ -0,0 +1,266 @@
+        members_data = request.env['mail.group.member'].sudo().search([('email_normalized', '=', email_normalized)])
+        members_data = {member.mail_group_id.id: True for member in members_data}
+
+        return request.render('mail_group.mail_groups', {
+            'mail_groups': mail_groups,
+            'messages_data': messages_data,
+            'members_data': members_data,
+            'email': email,
+        })
+
+    @http.route('/groups/subscription', type='json', auth='public', website=True)
+    def groups_subscription(self, group_id=0, action='subscribe', email=None, **kw):
+        """Subscribe the current logged user or the given email address to the mailing list.
+
+        If the user is logged, the action is automatically done.
+
+        But if the user is not logged (public user) an email will be send with a token
+        to confirm the action.
+
+        :param group_id: Id of the group
+        :param action: Action to perform (subscribe or unsubscribe)
+        :param email: Email to add in the member list
+        :return:
+            'done'
+                if the action has been done immediately
+            'email_sent'
+                if we send a confirmation email
+            'is_already_member'
+                if we try to subscribe but we are already member
+            'is_not_member'
+                if we try to unsubscribe but we are not member
+        """
+        if action not in ('subscribe', 'unsubscribe'):
+            raise werkzeug.exceptions.NotFound()
+
+        group = request.env['mail.group'].browse(int(group_id)).exists().sudo()
+        if not group:
+            raise werkzeug.exceptions.NotFound()
+
+        is_member = request.env['mail.group.member'].sudo().search_count([
+            ('mail_group_id', '=', group_id),
+            ('email_normalized', '=', tools.email_normalize(email)),
+        ])
+        if action == 'subscribe' and is_member:
+            return 'is_already_member'
+        elif action == 'unsubscribe' and not is_member:
+            return 'is_not_member'
+
+        if self._is_user_logged():
+            # For logged user, automatically join / leave without sending a confirmation email
+            if action == 'subscribe':
+                group._join_group(request.env.user.email, request.env.user.partner_id.id)
+            else:
+                group._leave_group(request.env.user.email)
+
+            return 'done'
+
+        else:
+            # For non-logged user, send an email with a token to confirm the action
+            group._send_confirmation_email(email, action)
+            return 'email_sent'
+
+    @http.route([
+        '/groups/<model("mail.group"):group>',
+        '/groups/<model("mail.group"):group>/page/<int:page>',
+    ], type='http', auth='public', sitemap=True, website=True)
+    def groups_messages(self, group, page=1, mode='thread', date_begin=None, date_end=None, **post):
+        Message = request.env['mail.group.message']
```

Comment:
Same here, would somehow add some group-based security and check access is granted before going deeper into message analysis / fetch.

Reply:
Done automatically with the access rules